### ISSUE TITLE
Default exported named class throw an error if there is another class with same name

### ISSUE THREAD

**Issue by cevek** [microsoft/TypeScript#16598]
**TypeScript Version:**  2.4.0 / nightly (2.5.0-dev.201xxxxx)

**Code**

```ts
class A {

}
export default class A {

}
```

**Expected behavior:**
No error

**Actual behavior:**
A module cannot have multiple default exports

**It's ok**
```ts
class A {

}
var y = class A {

}
export default y;
```

**Comment by weswigham** [comment]
This is a case where we're just reporting the less preferable error message - we should be reporting `Can not reassign block-scoped variable 'A'.` (since both `export default class A {}` and `class A {}` attempt to bind a block-scoped `A`). It is also worth mentioning that
```ts
var y = class A {}
export default y;
```
and
```ts
export default class A {}
```
are meaningfully different - the first makes a binding `y` in the file, while the second makes a binding `A` - the correct comparison in the original post would be to
```ts
class A {}
let A = class A {}
export default A;
```
which we (correctly) issue a duplicate identifier error on.

**Comment by mhegazy** [comment]
The behavior is correct, as @weswigham noted. this is per the ES6 spec, `export default` target is a a `HoistableDeclaration` so it is visible throughout the module. the error could be more expressive. though our other error here is `duplicate declaration`, so the bar is fairly low.

**Comment by RyanCavanaugh** [comment]
The current error printback is
```
a.ts:1:7 - error TS2528: A module cannot have multiple default exports.

1 class A {}
        ~

  a.ts:2:22 - Another export default is here.
    2 export default class A {}
                           ~
a.ts:2:22 - error TS2528: A module cannot have multiple default exports.

2 export default class A {}
                       ~

  a.ts:1:7 - The first export default is here.
    1 class A {}
            ~

Found 2 errors in the same file, starting at: a.ts:1
```
which feels sufficient to debug the problem even if the error message is somewhat misleading. Given the rarity of making this kind of mistake, this doesn't seem to merit further attention.
